### PR TITLE
Running End-to-End tests on SauceLabs

### PR DESCRIPTION
**Pull request by cp4r3z**
## Summary

Relating to #610, this PR integrates the "hello spoke" test with Sauce Labs. Instructions we added for running tests against a local browser and the Sauce Labs' Selenium grid.

## Developer notes

1. This PR will require secure strings (notably the auth0 and SauceLabs access keys) that will be stored as Travis-CI environment variables. **Merge checks will fail until that happens.**
2. Still haven't implemented the custom reporter to Spoke for passed/failed status of e2e tests
3. Travis-CI will be improved a bit in a future branch. Namely cleaning up the output, eliminating the "dumb" wait after the server launches, and perhaps running the tests in a matrix

### CODE REVIEW

**Comment by cp4r3z** [comment]
This PR will require secure strings (notably the auth0 and SauceLabs access keys) that will be stored as Travis-CI environment variables.

As you can see, I did make an attempt at using JWT (private key encryption tokens) but it was a bit complicated and Travis-CI provides a way of storing and deploying secure variables.

See in action here:
https://travis-ci.org/cp4r3z/Spoke/builds/392981194

image

So, prior to this merge, I need someone from the Spoke team to help add some env vars to your Travis-CI config. These variables are not visible to anyone, even after they're created.

@shakalee14 Would you like me to add the required variables to our Google Doc?

**Comment by shakalee14** [comment]
since i merged #642, can you rebase to resolve conflicts? and i can update the Travis-CI environment variables

File: jest.config.e2e.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+const overrides = {
+  setupTestFrameworkScriptFile: '<rootDir>/__test__/e2e/util/setup.js',
+  testMatch: ['**/__test__/e2e/**/*.js'],
+  testPathIgnorePatterns: ['<rootDir>/node_modules/', '<rootDir>/__test__/e2e/util/', '<rootDir>/__test__/e2e/pom/']
+}
+const merges = {
+  // Merge in changes to deeper objects
+  globals: {
+    // This sets the BASE_URL for the target of the e2e tests (what the tests are testing)
+    BASE_URL: 'localhost:3000'
```
Comment (shakalee14) [inline jest.config.e2e.js]: should this pull from an .env?

Comment (cp4r3z) [reply jest.config.e2e.js]: I thought the same and I tried. However, the test (client) process isn't the same as the target (server) process, so none of the .env variables are available to the test. I suppose we could look into adding a foreman file but it's yet another piece of the config puzzle.

**Comment by cp4r3z** [comment]
@shakalee14 Yes, I noticed that, and maybe we could talk it through sometime early this week? I don't like seeing red :-(

It's timing out when it attempts to instantiate the Selenium driver. I tried looking through my SauceLabs account for helpful logs, but I didn't see anything obvious. I've posted my SauceLabs password to the Google Doc if you want to check for yourself.

My current theory: SauceLabs is smart enough to know which repo you're trying to test and my SauceLabs account isn't authorized to test MoveOnOrg/Spoke. My theory assumes that you used my SauceLabs user/key in the MoveOnOrg Travis env vars. <- Is that true?

**Comment by shakalee14** [comment]
Great @cp4r3z - the tests are passing now - can we make sure this doesn't run on PRs coming from outside forks/contributors who don't have write access to the repo (so it won't automatically fail)

**Comment by cp4r3z** [comment]
@shakalee14 Yeah, have a look at the script: Spoke/travis-run-e2e-tests
```
if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then
...
```
It's been running for the last week on my wip branch and skips the e2e tests for forked PRs. Example:
https://travis-ci.org/MoveOnOrg/Spoke/builds/400016054

I guess my only hesitation is polluting the root folder. Maybe it ought to be placed in dev-tools or e2e. I figured it belonged at the root with the jest scripts.

**Comment by shakalee14** [comment]
@cp4r3z We can merge - just need to update readme to point to new SL acct. And we can clean up revisit clean up later.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 [![Build Status](https://travis-ci.org/MoveOnOrg/Spoke.svg?branch=main)](https://travis-ci.org/MoveOnOrg/Spoke)
 
+[![Sauce Test Status](https://saucelabs.com/buildstatus/cp4r3zsl)](https://saucelabs.com/u/cp4r3zsl)
```
Comment (shakalee14) [inline README.md]: update to the following --> https://saucelabs.com/u/opensourcemoveon

**Review by shakalee14 — APPROVED** [review]

**Pull request merged**